### PR TITLE
GLTF Importer v0

### PR DESCRIPTION
**Pull request by buddhaman12**
GLTF Importer works with a simple cube example.

### CODE REVIEW

**Review by ThiDaGui — APPROVED** [review]

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 1]
```diff
@@ -1,26 +1,623 @@
...
+  if (
+    mode === MeshPrimitiveModes.LINE_STRIP ||
+    mode === MeshPrimitiveModes.TRIANGLE_STRIP
+  ) {
+    cellSize = count;
+  } else if (mode === MeshPrimitiveModes.LINE_LOOP) {
+    cellSize = count + 1;
+  }
+
+  // let nCells = GetNumberOfCellsForPrimitive(mode, numberOfCoponents, count)
```
Comment (ThiDaGui) [inline Sources/IO/Geometry/GLTFImporter/index.js:116]: deadcode

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 2]
```diff
@@ -1,26 +1,623 @@
...
```
Comment (ThiDaGui) [inline Sources/IO/Geometry/GLTFImporter/index.js:179]: deadcode

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 3]
```diff
@@ -1,26 +1,623 @@
...
```
Comment (ThiDaGui) [inline Sources/IO/Geometry/GLTFImporter/index.js:242]: deadcode

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 4]
```diff
@@ -1,26 +1,623 @@
...
+        const actor = buildActorFromPrimitive(primitive);
+
+        model.renderer.addActor(actor);
+
+        actor.getMapper().setScalarVisibility(true);
+        const clr = { r: 200 / 255.0, g: 200 / 255.0, b: 200 / 255.0 };
+        actor.getProperty().setColor(clr.r, clr.g, clr.b);
+      });
+
+      if (node.children != null) {
```
Comment (ThiDaGui) [inline Sources/IO/Geometry/GLTFImporter/index.js:553]: what happens if we have more than one level in the node hierarchy ?

**Review by ThiDaGui — APPROVED** [review]
looks good for a V0

File: Sources/IO/Geometry/GLTFImporter/index.d.ts [hunk 5]
```diff
@@ -14,24 +16,24 @@ type vtkGLTFImporterBase = vtkObject & Omit<vtkAlgorithm,
...
+	| 'setOuputConnection'
+	| 'getOutputConnection'>;
 
 export interface vtkGLTFImporter extends vtkGLTFImporterBase {
-  	/**
-	 * Get the url of the object to load.
-	 */
-	getUrl(): string;
+  /**
+  * Get the url of the object to load.
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.d.ts:26]: wrong indent

File: Sources/IO/Geometry/GLTFImporter/index.d.ts [hunk 6]
```diff
@@ -14,24 +16,24 @@ type vtkGLTFImporterBase = vtkObject & Omit<vtkAlgorithm,
...
+  */
+  getUrl(): string;
+
+  setUrl(url: string): Promise<void>;
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.d.ts:30]: please document all methods

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 7]
```diff
@@ -1,26 +1,623 @@
...
+  const indicesCellArray = vtkCellArray.newInstance();
+
+  const currentCell = Array(cellSize);
+  const accessorBegin = byteOffset;
+  const accessorEnd = accessorBegin + count * byteStride;
+
+  let val;
+
+  if (mode === MeshPrimitiveModes.TRIANGLE_FAN) {
+    let i = 0;
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.js:127]: does not seem needed

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 8]
```diff
@@ -1,26 +1,623 @@
...
+
+      if (it <= accessorBegin + byteStride) {
+        i++;
+      } else {
+        indicesCellArray.insertNextCell(currentCell);
+        currentCell[1] = currentCell[2];
+      }
+    }
+  } else {
+    let i = 0;
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.js:140]: rename `i` to `cellPointIndex` or something like that ?

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 9]
```diff
@@ -1,26 +1,623 @@
...
+      }
+    }
+
+    if (mode === MeshPrimitiveModes.LINE_LOOP) {
+      currentCell[currentCell.length - 1] = currentCell[0];
+      indicesCellArray.insertNextCell(currentCell);
+    }
+  }
+  return indicesCellArray;
+}
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.js:284]: can't `extractDataArrayFloat`, `extractCellBufferDataUint32`, `extractCellBufferDataUint16` and `extractCellBufferDataUint8` be factorized ?
Note that you can do `inputBuffer[aMethodAsString](it, true)` where `aMethodAsString==="getUint32"`

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 10]
```diff
@@ -1,26 +1,623 @@
...
+        uri = model.baseURL.concat('/', uri);
+      }
+
+      return fetchData(uri, { binary: true }).then((bufferContent) => {
+        const dataView = new DataView(bufferContent);
+        model.GLTFData.buffers[index].dataView = dataView;
+        return dataView;
+      });
+    });
+    await Promise.all(promises);
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.js:517]: add a comment to explain why you need to wait for all the fetches to be done before going to the next step

File: Sources/IO/Geometry/GLTFImporter/index.js [hunk 11]
```diff
@@ -1,26 +1,623 @@
...
+    case AccessorComponentTypes.FLOAT:
+      return extractDataArrayFloat(
+        buffer,
+        byteOffset,
+        bufferView.byteStride ?? 4 * numberOfComponents,
+        accessor.count,
+        numberOfComponents
+      );
+    default:
+      throw new Error('unsupported accessor type for primitive attribute');
```
Comment (finetjul) [inline Sources/IO/Geometry/GLTFImporter/index.js:336]: I see a lot of throw in your code but without catch

**Pull request merged**